packages/repository/src/repositories/constraint-utils.ts

```diff
@@ -55,6 +55,7 @@ export function constrainDataObject<T extends Entity>(
 ): DataObject<T> {
   const constrainedData = cloneDeep(originalData);
   for (const c in constraint) {
+    if (constrainedData[c]) throw new Error(`Field "${c}" cannot be changed!`);
```

Comment:
I think we use "property" rather than "field", would you mind rewording this message for consistency?

Reply:
Also, this check is not going to catch the case where `constrainedData[c]` is set to a falsey value like `0` or `null`. Please fix, e.g. by using the following condition: `if (c in constrainedData)`.

Reply:
good catch!